src/Raven.Client/Documents/Operations/AttachmentResult.cs

```diff
@@ -1,12 +1,40 @@
-﻿namespace Raven.Client.Documents.Operations
+﻿using System;
+using System.IO;
+using System.Net.Http;
+
+namespace Raven.Client.Documents.Operations
 {
-    public class AttachmentResult
+    public class AttachmentResultWithStream : AttachmentResult, IDisposable
+    {
+        private HttpResponseMessage _response;
+
+        public Stream Stream;
+
+        public AttachmentResultWithStream(HttpResponseMessage response)
+        {
+            _response = response;
+        }
+
+        public void Dispose()
+        {
+            Stream?.Dispose();
+            Stream = null;
+            _response?.Dispose();
+            _response = null;
+        }
+    }
+
+    public class AttachmentResult : AttachmentName
```

Comment:
It is worth a separate class just for that? Can just return (AttachmentNAme, etag)